cli/src/main/java/org/adamalang/cli/devbox/RxPubSub.java

```diff
@@ -0,0 +1,39 @@
+package org.adamalang.cli.devbox;
+
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.adamalang.api.DataResponder;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class RxPubSub {
+    private final AtomicInteger currentId;
+    public final ConcurrentHashMap<Integer, DataResponder> responders;
+    public final static RxPubSub instance = new RxPubSub();
+
+    public RxPubSub() {
+        currentId = new AtomicInteger(1);
+        this.responders = new ConcurrentHashMap<>();
+    }
+
+    public Integer getNextId() {
+        return currentId.getAndIncrement();
+    }
+
+    public void subscribe(Integer id, DataResponder responder) {
+        responders.put(id, responder);
+    }
+
+    public void unsubscribe(Integer id) {
+        if (responders.containsKey(id)) {
+            responders.remove(id);
+        }
+    }
```

Comment:
This is conceptually leaky and can be simplified with a "Cancellation" based model.

public Runnable subscribe(DataResponder) {
  int id = currentId.getAndIncrement();
  responders.put(id, responder);
  return () -> responders.remove(id);
}

Now, for language, I've thought about just created a Cancelable interface common, but I've just been using Runnable.